src/extensions/surveys.tsx

```diff
@@ -444,6 +445,51 @@ export function usePopupVisibility(
         }
     }, [])
 
+    // Add URL change listener to hide survey when URL no longer matches
+    useEffect(() => {
+        if (isPreviewMode || !survey.conditions?.url || !posthog) {
```

Comment:
do we need `posthog`? its not being used in this block

Reply:
removed, thanks for pointing